src/Init/Prelude.lean

```diff
@@ -66,6 +66,13 @@ example (b : Bool) : Function.const Bool 10 b = 10 :=
 @[inline] def Function.const {α : Sort u} (β : Sort v) (a : α) : β → α :=
   fun _ => a
 
+/--
+The encoding of `let_fun x := v; b` is `letFun v (fun x => b)`.
+This is an abbreviation for `(fun x => b) v`, so the value of `x` is not accessible to `b`,
+unlike with `let x := v; b`.
+-/
+def letFun {α : Sort u} {β : α → Sort v} (v : α) (f : (x : α) → β x) : β v := f v
```

Comment:
This encoding means that even with `config.zeta` false, we reduce `let_fun`s when at default reducibility, correct? Is that intentional or should it be `[irreducible]`?

Reply:
Good point, that is something I did not think enough about. I made it be irreducible and added more tests. (I also extended the optimization for ignoring unnecessary lets in `isDefEqQuick` to also consider `let_fun`.)

Reply:
@leodemoura I'll wait for your final confirmation on this point before merging

Reply:
Agreed. We don't want to reduce `let_fun`s when `config.zeta = false` and using the default reducibility.

Reply:
Agreed. We don't want to reduce `let_fun`s when `config.zeta = false` and using the default reducibility.